js/app.js

```diff
@@ -48,10 +50,18 @@ $(document).ready(function() {
 
   // Input binding
   $searchInput
-  .on('keyup', function() {
-    var query = $(this).val();
-    toggleIconEmptyInput(query);
-    algoliaHelper.setQuery(query).search();
+  .on('keyup paste cut change', function() {
+    var $el = $(this);
+
+    // We use setTimeout to defer the callback to the bottom of the stack to
+    // catch the value after the paste, cut, or delete via mouse
+    setTimeout(function(){
```

Comment:
@bobylito Great point about getting it off the event parameter. Perfect optimization. The `clipboardData` won't work because they might be pasting on to an existing search query, but paired with @vvo's suggestion below the event param works like a charm. :+1: